proxy/proxy.go

```diff
@@ -213,13 +209,12 @@ func (p *Proxy) Handle(w http.ResponseWriter, r *http.Request) {
 				} else if cookie.Name == CookieJenkinsIdled { //Found a cookie saying Jenkins is idled, verify and act accordingly
 					if c, ok := p.ProxyCache.Get(cookie.Value); ok {
 						pci := c.(ProxyCacheItem)
-						oc := ic.NewOpenShiftWithClient(http.DefaultClient, pci.ClusterURL, pci.OsoToken)
-						isIdle, err := oc.IsIdle(pci.NS, ServiceName)
+						isIdle, err := p.idler.IsIdle(pci.NS)
```

Comment:
+1. So, for now, we really just need the IsIdle call. Make things imo already clearer. _ProxyCacheItem_ Still has a _OsoToken_. Is that still needed for something? If not we can remove it, together with any helpers (_GetOSOToken_ for example). The less the better and not using the OSO token reduces the amount of places where it is used and hence potentially exposed.

Reply:
Now I am also more free to continue my refactoring on the Idler. Effectively all becomes _internal_.

Reply:
We actually need it to log in to Jenkins, but I need to check if we need to cache it - that might not be needed, let me check

Reply:
ok